addons/viewport/src/components/Panel.jsx

```diff
@@ -0,0 +1,120 @@
+/* global document */
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { viewports, defaultViewport, resetViewport } from './viewportInfo';
+
+import { SelectViewport } from './SelectViewport';
+import { RotateViewport } from './RotateViewport';
+
+import * as styles from './styles';
+
+const storybookIframe = 'storybook-preview-iframe';
+const containerStyles = {
+  padding: 15,
+  width: '100%',
+  boxSizing: 'border-box',
+  fontFamily:
+    '-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", Arial, sans-serif',
+};
+
+export class Panel extends Component {
+  static propTypes = {
+    channel: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
+  };
+
+  constructor(props, context) {
+    super(props, context);
+    this.state = {
+      viewport: defaultViewport,
+      isLandscape: false,
+    };
+
+    this.props.channel.on('addon:viewport:update', this.changeViewport);
+  }
+
+  componentDidMount() {
+    this.iframe = document.getElementById(storybookIframe);
+  }
+
+  componentWillUnmount() {
+    this.props.channel.removeListener('addon:viewport:update', this.changeViewport);
+  }
+
+  iframe = undefined;
```

Comment:
I would prefer to set this in the constructor

Reply:
Is it a general suggestion not to use property initializers? Why? Maybe we should setup a eslint rule then?